sdk/core/core/test/cmocka/test_az_credential_client_secret.c

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "az_credentials.h"
```

Comment:
What changed in this file? I am asking to know what to review. If it was just a file rename, why didn't git just detect it as a rename from `test_az_aad.c` to `test_az_credential_client_secret.c`.

Reply:
It is a new test. And I completely removed the old one. I would not consider it a rename, as I wrote this test from scratch. Previous code did not check the client secret credential end to end, just the add part. It did not check whether the credential applies auth header to the request, it did not check whether the credential handles retriable errors, it did not check whether the credential will reuse the token when it did not expire.

Reply:
Gotcha. Thanks.

Will the test now fail if we change the following line to catch incorrectly created body?

https://github.com/Azure/azure-sdk-for-c/blob/04ca0526cf798513d9640a345eedea664429aec9/sdk/core/core/src/az_aad.c#L138
 ```diff
-*out_body = az_span_slice(body, 0, _az_span_diff(remainder, body));
+*out_body = az_span_slice(body, 0, url_length);
```

Reply:
Yes, I just confirmed that, it catches the bug.